Fix issue with plugin list download

bruderstein opened the pull request:
When the plugin list matched the MD5 (which it hasn't done for a while due to a server side issue with the MD5 not being correct!), it would always fail.

The server side issue was corrected, which manifested this issue in the code.

Also, the DEV list MD5 URL wasn't used when the DEV list was selected.

This is the fix for #57

bruderstein commented on pluginManager/src/PluginList.cpp:
```diff
@@ -904,18 +904,21 @@ void PluginList::downloadList()
-	if (downloadResult && serverMD5 != cHashBuffer.get())
+	if (downloadResult && serverMD5 == cHashBuffer.get()) {
+		// Server hash matches local hash, so we're ok to continue
+		downloadSuccess = TRUE;
```
This is the actual fix. If the hash matched with the server, it would error later.

chcg replied:
And I already wanna suggest to add a checksum onto the complete list to avoid unnecesary downloads but it seems this was already there.

Could we also find out that the downloaded list is corrupted? Like seen in my post to #57, where it just contained the HTML error page?

chcg commented:
Wow, just wanna comment on it, but you were too fast. Would have been:

> Maybe:
> - a timing related issue not observed so far and now showing up more often.
> - a not initialized value, but that would probably be more sporadic
> - threading synchronization issue
> 
> , maybe need to be rechecked with 1.4.3, now the server is up again and see if 


How could the issue be explained for the standard list as mentioned in the answer of #57? That one I'm not facing.

bruderstein commented:
I've changed the server hash and cleared the caches, so it won't be an issue now. There's very aggressive caching on the prod instance, so it wouldn't have kicked in until all the edge caches have the updated hash which matched the contents. It was only when we updated the list to v1.4.5 that the server md5 issue was noticed and fixed, which triggered this issue (just slowly, because of the caching)

bruderstein commented:
Good catch CPPcheck :) I'll push it in a separate PR.

criskolkman commented:
It doesn't seem to be fixed. Just did a fresh install of Notepad++ and added Plugin Manager (shame you have to do it yourself nowadays), but Plugin Manager is not able to get the plugin list.
